Sources/GRPCCore/Metadata.swift

```diff
@@ -14,5 +14,329 @@
+  }
+
+  /// Add a new key-value pair, where the value is binary data, in the form of `Data`.
+  /// - Parameters:
+  ///   - binaryValue: The `Data` to be associated with the given key.
+  ///   - key: The key to be associated with the given value. Must end in `-bin`.
+  public mutating func addBinary(_ binaryValue: [UInt8], forKey key: String) {
+    self.addValue(.binary(binaryValue), forKey: key)
+  }
+
+  /// Add a new key-value pair.
+  /// - Parameters:
+  ///   - value: The ``Value`` to be associated with the given key.
+  ///   - key: The key to be associated with the given value. If value is binary, it must end in `-bin`.
+  public mutating func addValue(_ value: Value, forKey key: String) {
+    self.elements.append(.init(key: key, value: value))
+  }
+
+  /// Removes all values associated with the given key.
+  /// - Parameter key: The key for which all values should be removed.
+  public mutating func removeAllValues(forKey key: String) {
+    elements.removeAll { metadataKeyValue in
+      metadataKeyValue.key == key
+    }
+  }
+
+  /// Adds a key-value pair to the collection, where the value is a string.
+  /// If there are pairs already associated to the given key, they will all be removed first, and the new pair
+  /// will be added. If no pairs are present with the given key, a new one will be added.
+  /// - Parameters:
+  ///   - stringValue: The string value to be associated with the given key.
+  ///   - key: The key to be associated with the given value.
+  public mutating func replaceOrAddString(_ stringValue: String, forKey key: String) {
+    self.replaceOrAddValue(.string(stringValue), forKey: key)
+  }
+
+  /// Adds a key-value pair to the collection, where the value is `Data`.
+  /// If there are pairs already associated to the given key, they will all be removed first, and the new pair
+  /// will be added. If no pairs are present with the given key, a new one will be added.
+  /// - Parameters:
+  ///   - binaryValue: The `Data` to be associated with the given key.
+  ///   - key: The key to be associated with the given value. Must end in `-bin`.
+  public mutating func replaceOrAddBinary(_ binaryValue: [UInt8], forKey key: String) {
+    self.replaceOrAddValue(.binary(binaryValue), forKey: key)
+  }
+
+  /// Adds a key-value pair to the collection.
+  /// If there are pairs already associated to the given key, they will all be removed first, and the new pair
+  /// will be added. If no pairs are present with the given key, a new one will be added.
+  /// - Parameters:
+  ///   - value: The ``MetadataValue`` to be associated with the given key.
+  ///   - key: The key to be associated with the given value. If value is binary, it must end in `-bin`.
+  public mutating func replaceOrAddValue(_ value: Value, forKey key: String) {
+    self.removeAllValues(forKey: key)
+    elements.append(.init(key: key, value: value))
+  }
+}
+
+extension Metadata: RandomAccessCollection {
+  public typealias Element = KeyValue
```

Comment:
Can we spell out the tuple here? Typealiases make it harder for users to see the actual type. We can use the typealias internally though.

Reply:
This was the only place I was using the typealias actually so I got rid of it